### ISSUE TITLE
deploying elastic/eck-operator helm chart on OpenShift 4.11 fails with exec: \"./elastic-operator\": stat ./elastic-operator: permission denied

### ISSUE THREAD

**Issue by cohenuzi01** [elastic/cloud-on-k8s#6017]
When deploying elastic/eck-operator helm chart (eck-operator-2.4.0 helm chart, helm repo: https://helm.elastic.co/) on OpenShift 4.11, deployment fails, getting:
msg="runc create failed: unable to start container process: exec: \"./elastic-operator\": stat ./elastic-operator: permission denied"
and the pod fails to start.
Note that deploying the elastic/eck-operator helm chart on OpenShift 4.9 and 4.10 works (so this issue is specific to OpenShift 4.11)

helm reports the following warning (but then adds the missing properties to the container securityContext , see below):
```
helm repo add elastic https://helm.elastic.co/
helm install elastic-operator elastic/eck-operator -n logging --set=installCRDs=false --set=webhook.enabled=false   --set=config.validateStorageClass=false

W0915 14:52:29.426212   70832 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "manager" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "manager" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "manager" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

NAME: elastic-operator
LAST DEPLOYED: Thu Sep 15 14:52:27 2022
NAMESPACE: logging
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Inspect the operator logs by running the following command:
   kubectl logs -n logging sts/elastic-operator
```

Pod fails to create a container:
```
$ k get pods -n logging
NAME                 READY   STATUS                 RESTARTS   AGE
elastic-operator-0   0/1     CreateContainerError   0          2m21s
```

Running kubectl describe pods -n logging elastic-operator-0 I see the following event message:
```
  Warning  Failed          71s                   kubelet            Error: container create failed: time="2022-09-15T18:53:49Z" level=error msg="runc create failed: unable to start container process: exec: \"./elastic-operator\": stat ./elastic-operator: permission denied"
  Warning  Failed          32s (x3 over 59s)     kubelet            (combined from similar events): Error: container create failed: time="2022-09-15T18:54:28Z" level=error msg="runc create failed: unable to start container process: exec: \"./elastic-operator\": stat ./elastic-operator: permission denied"
```

The pod starts using:
      Service Account:  elastic-operator and:
      openshift.io/scc: restricted-v2
      seccomp.security.alpha.kubernetes.io/pod: runtime/default

The pod gets scheduled, so it's not an scc issue.
The pod 's securityContext is:
 ```
   securityContext:
      fsGroup: 1000670000
      runAsNonRoot: true
      seLinuxOptions:
        level: s0:c26,c10
      seccompProfile:
        type: RuntimeDefault
```
The container's securityContext is:
```
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
          - ALL
        runAsUser: 1000670000
```

Any idea why the container cannot execute ./elastic operator

**Environment**
RedHat OpenShift 4.11 (running on 
 (Kubernetes Server Version: v1.24.0+4f0dd4d)

* ECK version:
helm chart:
NAME                     	CHART VERSION	APP VERSION
elastic/eck-operator     	2.4.0        	2.4.0
image:
docker.elastic.co/eck/eck-operator:2.4.0

```
$  kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.12", GitCommit:"b058e1760c79f46a834ba59bd7a3486ecf28237d", GitTreeState:"clean", BuildDate:"2022-07-13T14:59:18Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0+4f0dd4d", GitCommit:"0a57f1f59bda75ea2cf13d9f3b4ac5d202134f2d", GitTreeState:"clean", BuildDate:"2022-08-10T15:36:44Z", GoVersion:"go1.18.4", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.22) and server (1.24) exceeds the supported minor version skew of +/-1

```
Opened also in discuss.elastic.co
Ref: https://discuss.elastic.co/t/unable-to-deploy-elastic-eck-operator-on-redhat-openshift-4-11/314534

**Comment by barkbay** [comment]
I'm able to reproduce the `stat ./elastic-operator: permission denied"` on OCP `4.11.6` with the actual ECK `2.4.0` release.
However, using a snapshot Docker image of the upcoming `2.5.0` release the operator starts as expected.
The only change in the Docker image I can think of at the moment is https://github.com/elastic/cloud-on-k8s/pull/5983/files, but I don't understand why it would solve this particular problem 🤔

**Comment by barkbay** [comment]
I'm not 100% sure about the root cause of the permission error, however:
* The next release (`2.5.0`) does not seem to be affected.
* The UBI version of the ECK `2.4.0` image seems to run as expected, so as a workaround I would suggest to try the following:
```
helm install elastic-operator elastic/eck-operator -n logging \
                   --set=installCRDs=false --set=webhook.enabled=false \
                   --set=config.validateStorageClass=false \
                   --set=image.repository=docker.elastic.co/eck/eck-operator-ubi8
```

I'll create a PR regarding the `PodSecurity` warnings .

**Comment by barkbay** [comment]
Related PR for the warnings: https://github.com/elastic/cloud-on-k8s/pull/6047

**Comment by barkbay** [comment]
Added the `bug` label because it should be possible to run the operator with the aforementioned `securityContext`.
As already mentioned this issue should be solved in the next release (`2.5.0`), and the workaround for now is to use the UBI image.
